### PR TITLE
[CT-641] Update deleveraging handler for DeleveragingEventV1

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
Protocol is currently sending quote quantums as the price (fill size * unit price) in DeleveragingEventV1 [here](https://github.com/dydxprotocol/v4-chain/blob/b541e28be6d0dea0431243a372f781fdb70265bb/protocol/x/clob/keeper/deleveraging.go#L403).
In the handler, to correctly handle this event, quoteAmount should be set to event_data->price, and price should be set to quoteAmount/size.
Also, add additional validation that the fillAmount/price of the deleveraging event is not 0.

### Test Plan
Unit tested.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/ender/src/scripts/handlers/dydx_deleveraging_handler.sql [hunk 1]
```diff
@@ -55,8 +56,9 @@ BEGIN
     */
     size = dydx_trim_scale(dydx_from_jsonlib_long(event_data->'fillAmount') *
                                  power(10, perpetual_market_record."atomicResolution")::numeric);
-    price = dydx_trim_scale(dydx_from_jsonlib_long(event_data->'price') *
+    quote_amount = dydx_trim_scale(dydx_from_jsonlib_long(event_data->'price') *
                                   power(10, QUOTE_CURRENCY_ATOMIC_RESOLUTION)::numeric);
+    price = dydx_trim_scale(quote_amount / size);
```
Comment (vincentwschau) [inline indexer/services/ender/src/scripts/handlers/dydx_deleveraging_handler.sql:61]: nit: Add a comment indicating the event is incorrect and that's why this math is being done.

**Comment by vincentwschau** [comment]
I think we actually can change the name of the field btw, from price -> total_amount or something and it shouldn't break how the proto is encoded / decoded to make the code more readable rather than sprinkling in comments everywhere. Non-blocking though for this review.

**Comment by vincentwschau** [comment]
As part of this change, can you also add a TODO to the line of code in the protocol where the total amount deleveraged is incorrectly set as the price the deleveraging occurred at? Would be good to have a record.

**Review by vincentwschau — APPROVED** [review]

File: indexer/services/ender/__tests__/handlers/order-fills/deleveraging-handler.test.ts [hunk 2]
```diff
@@ -229,9 +229,9 @@ describe('DeleveragingHandler', () => {
     );
 
     // This size should be in fixed-point notation rather than exponential notation.
-    const quoteAmount: string = '0.00000000001'; // quote amount is price * fillAmount = 1e3 * 1e-14 = 1e-11
+    const quoteAmount: string = '1000'; // quote amount is event->price * QUOTE_CURRENCY_ATOMIC_RESOLUTION = 1e9*1e-6=1e3
```
[the remark this answers is not in the document]
Comment (dydxwill) [reply indexer/services/ender/__tests__/handlers/order-fills/deleveraging-handler.test.ts:232]: I just updated the field name to totalQuoteQuantums as you suggested instead.

File: indexer/services/ender/src/scripts/handlers/dydx_deleveraging_handler.sql (hunk 1 above)
Comment (dydxwill) [reply indexer/services/ender/src/scripts/handlers/dydx_deleveraging_handler.sql:61]: I just updated the field name to totalQuoteQuantums as you suggested instead.

File: indexer/services/ender/__tests__/handlers/order-fills/deleveraging-handler.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vincentwschau) [reply indexer/services/ender/__tests__/handlers/order-fills/deleveraging-handler.test.ts:232]: Comment probably should be updated to not refer to `event->price` then.

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**